packages/core/src/shared/filetypes.ts

```diff
@@ -102,27 +117,231 @@ export function activate(): void {
                 await vscode.languages.setTextDocumentLanguage(doc, 'ini')
             }
 
-            if (await isSameMetricPending(telemKind, fileExt)) {
-                return // Avoid redundant/duplicate metrics.
-            }
-
-            telemetry.file_editAwsFile.emit({
+            const metric: FileEditAwsFile = {
                 awsFiletype: telemKind,
                 passive: true,
                 result: 'Succeeded',
                 filenameExt: fileExt,
-            })
+            }
+            const isSameMetricPending = await globals.telemetry.findPendingMetric('file_editAwsFile', metric, [
+                'awsFiletype',
+                'filenameExt',
+            ])
+            if (isSameMetricPending) {
+                return // Avoid redundant/duplicate metrics.
+            }
+
+            telemetry.file_editAwsFile.emit(metric)
         }, undefined)
     )
 }
 
-async function isSameMetricPending(filetype: string, fileExt: string | undefined): Promise<boolean> {
-    const pendingMetrics = await collectionUtil.first(
-        globals.telemetry.findIter((m) => {
-            const m1 = m.Metadata?.find((o) => o.Key === 'awsFiletype')
-            const m2 = m.Metadata?.find((o) => o.Key === 'filenameExt')
-            return m.MetricName === 'file_editAwsFile' && m1?.Value === filetype && m2?.Value === fileExt
-        })
-    )
-    return !!pendingMetrics
+/**
+ * Most common programming langauges and their file extensions
+ * Includes list from https://survey.stackoverflow.co/2023/#technology plus extra items.
+ */
+export const codefileExtensions = new Set([
```

Comment:
non-blocking but I assume this list is a subset of the original file extension list, could you point out which extensions were added?

Reply:
it's a superset not a subset. i added the extensions specified here: https://taskei.amazon.dev/tasks/ff8709ad-fab5-467d-9ada-82ff4a2cd92f?activeTab=comments#a6552700-fc49-41a1-9af7-ac5c7d2b58a3

```
.bat
.cmake
.f03
.f08
.f77
.for
.gql
.graphql
.gs
.gsp
.gst
.gsx
.jsx
.mak
.makefile
.sb
.sb2
.sql
.tf
.vue
.xml
.yaml
```

Reply:
yeah oops, i meant superset. thanks for the info!